security/crypto/ecdsa/ecdsa.go

```diff
@@ -115,80 +110,86 @@ func (ec *ecdsaBase) Combine(signatures ...hotstuff.QuorumSignature) (hotstuff.Q
 // Verify verifies the given quorum signature against the message.
-func (ec *ecdsaBase) Verify(signature hotstuff.QuorumSignature, message []byte) bool {
+func (ec *ecdsaBase) Verify(signature hotstuff.QuorumSignature, message []byte) (err error) {
 	s, ok := signature.(crypto.Multi[*Signature])
 	if !ok {
-		ec.logger.Panicf("cannot verify signature of incompatible type %T (expected %T)", signature, s)
+		return fmt.Errorf("cannot verify signature of incompatible type %T (expected %T)", signature, s)
 	}
 	n := signature.Participants().Len()
 	if n == 0 {
-		return false
+		return fmt.Errorf("expected more than zero participants")
 	}
 
-	results := make(chan bool, n)
+	results := make(chan error, n)
 	hash := sha256.Sum256(message)
 	for _, sig := range s {
 		go func(sig *Signature, hash hotstuff.Hash) {
 			results <- ec.verifySingle(sig, hash)
 		}(sig, hash)
 	}
-	valid := true
 	for range s {
-		if !<-results {
-			valid = false
-		}
+		err = errors.Join(<-results)
+	}
+	if err != nil {
+		return err
 	}
-	return valid
+	return nil
 }
 
 // BatchVerify verifies the given quorum signature against the batch of messages.
-func (ec *ecdsaBase) BatchVerify(signature hotstuff.QuorumSignature, batch map[hotstuff.ID][]byte) bool {
+func (ec *ecdsaBase) BatchVerify(signature hotstuff.QuorumSignature, batch map[hotstuff.ID][]byte) (err error) {
 	s, ok := signature.(crypto.Multi[*Signature])
 	if !ok {
-		ec.logger.Panicf("cannot verify signature of incompatible type %T (expected %T)", signature, s)
+		return fmt.Errorf("cannot verify signature of incompatible type %T (expected %T)", signature, s)
 	}
 	n := signature.Participants().Len()
 	if n == 0 {
-		return false
+		return fmt.Errorf("expected more than zero participants")
 	}
 
-	results := make(chan bool, n)
+	results := make(chan error, n)
 	set := make(map[hotstuff.Hash]struct{})
 	for id, sig := range s {
 		message, ok := batch[id]
 		if !ok {
-			return false
+			return fmt.Errorf("message not found")
 		}
 		hash := sha256.Sum256(message)
 		set[hash] = struct{}{}
 		go func(sig *Signature, hash hotstuff.Hash) {
 			results <- ec.verifySingle(sig, hash)
 		}(sig, hash)
 	}
-	valid := true
 	for range s {
-		if !<-results {
-			valid = false
-		}
+		err = errors.Join(<-results)
+	}
+	if err != nil {
+		return err
 	}
-
 	// valid if all partial signatures are valid and there are no duplicate messages
-	return valid && len(set) == len(batch)
+	if len(set) != len(batch) {
+		return fmt.Errorf("invalid signature")
+	}
+	return nil
 }
 
-func (ec *ecdsaBase) verifySingle(sig *Signature, hash hotstuff.Hash) bool {
+func (ec *ecdsaBase) verifySingle(sig *Signature, hash hotstuff.Hash) error {
 	replica, ok := ec.config.ReplicaInfo(sig.Signer())
 	if !ok {
-		ec.logger.Warnf("ecdsaBase: got signature from replica whose ID (%d) was not in the config.", sig.Signer())
-		return false
+		return fmt.Errorf("ecdsaBase: got signature from replica whose ID (%d) was not in the config.", sig.Signer())
```

Comment:
Is it enough to just use `ecdsa:` instead of `ecdsaBase`?
What about:
```go
return fmt.Errorf("ecdsa: received signature from replica ID %d not in config", sig.Signer())
```
(note also, we don't use punctuation in error messages)